### PR TITLE
[DATAM-429] Remove Fixnum and replace with Integer per 2.4.1 Deperecations

### PR DESCRIPTION
**Pull request by kunday**
👻 
/cc @zendesk/bunyip-numbat 

### Description
At the moment we use Fixnum for certain numbers which will be removed in 2.4.2. Switching it to the recommended integer library. Also took the pleaserure to drop 2.0, 2.1 unsupported ruby versions and wwtd for local travis support. Thanks @grosser.

### References
- https://zendesk.atlassian.net/browse/DATAM-429

### Risks
Medium: Need to test in one project before making a release. ( likely batch )

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,10 +1,12 @@
 language: ruby
 sudo: false
 cache: bundler
+install:
+  bundle install
```
Comment (kunday) [inline .travis.yml]: Hd to do this explicitly after the container infrastructure blog post that travis guys sent out.

**Comment by kunday** [comment]
I'm trying to make a release of this gem - but i get this error
```
prop 2.2.3 built to pkg/prop-2.2.3.gem.
Tag v2.2.3 has already been created.
rake aborted!
Pushing gem to https://rubygems.org...
You do not have permission to push to this gem. Ask an owner to add you with: gem owner prop --add mail@kunday.com

Tasks: TOP => release => release:rubygem_push
(See full trace by running task with --trace)
(master ?:1 ✗) prop gem owner prop --add mail@kunday.com
Adding mail@kunday.com: You do not have permission to manage this gem.
(master ?:1 ✗) prop
```
Can some add me to the deployer list in rubygems or release this for me pretty please ? @grosser

File: .travis.yml (hunk 1 above)
Comment (grosser) [inline .travis.yml]: that breaks bundler caching ... I'll take a look later ... seems weird that it did not work out of the box ...

**Comment by grosser** [comment]
added you

File: .travis.yml (hunk 1 above)
Comment (grosser) [inline .travis.yml]: fixed in https://github.com/zendesk/prop/pull/28

**Pull request merged**